contributing-guides/style-guide.md

```diff
@@ -390,6 +390,42 @@ For example, `[d]ownload` in English may be translated into `[d]escargar` in Spa
 
 ## Example commands
 
+- The examples of pages in the subdirectory `common` for UNIX-like operating systems (e.g. `pages/common/tar.md`), preferably, should be written with the shell syntax specified by the [latest POSIX standard](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18).
+  However, if you believe the example is hard to read, has little to no use, or is impossible to write in POSIX-compliant syntax, use non-POSIX syntax and add a note or warning indicating it and some or all of its consequences concisely and specifically. For example:
+
+  ```md
+  # IFS
+
+  > IFS (Internal Field Separator) is a special environment variable that defines the delimiter used for word splitting in Unix shells.
+  > The default value of IFS is a space, tab, and newline. The three characters serve as delimiters.
+  > Note: Not all shells support ANSI C quoted strings (i.e., $'...' and $"..."), the dollar sign could be included in the string.
+  > More information: <https://www.gnu.org/software/bash/manual/html_node/Word-Splitting.html>.
+  ```
+
+  To help you decide using POSIX-compliant syntax, ask yourself this:
+
+  - Is the example too long (i.e. [exceeds the maximum line length](https://github.com/tldr-pages/tldr/blob/main/.markdownlint.json#L5))?
+  - Does the example utilize any [risky](https://unix.stackexchange.com/a/278439) or [inefficient](https://unix.stackexchange.com/a/169765) commands?
+  - Does the example return the correct output?
```

Comment:
```suggestion
  As more of the points below apply, so does the recommendation to not use POSIX-compliant syntax in a particular example:

  - The example is not present in at least one implementation
  - The example is too long (i.e. [exceeds the maximum line length](https://github.com/tldr-pages/tldr/blob/main/.markdownlint.json#L5))
  - The example utilize any [risky](https://unix.stackexchange.com/a/278439) or [inefficient](https://unix.stackexchange.com/a/169765) commands?
  - The example doesn't return the correct output in some or all implementations
```
How about this? @kbdharun @sebastiaanspeck @tricantivu

Reply:
IMO questions makes things a bit too open and a bit confusing (a positive answer means that I should use POSIX-compliant syntax or not?)

Reply:
> IMO questions makes things a bit too open and a bit confusing (a positive answer means that I should use POSIX-compliant syntax or not?)

Perhaps I could replace the questions for a guide explaining how to check POSIX-compliance (e.g. `shellcheck -s sh -`).